### ISSUE TITLE
Quarkus 1.11.2 to Quarkus 1.11.3 upgrade failure of a Quarkus Kafka project

### ISSUE THREAD

**Issue by missourian55** [quarkusio/quarkus#15089]
**Describe the bug**
Upgraded an existing resteasy, resteasy-jackson and Kafka project from 1.11.2 to 1.11.3 and application fail to start in dev mode

**Expected behavior**
Application should start similar to 1.11.2

**Actual behavior**
```java

	2021-02-15 14:56:57,678 INFO  [io.qua.dep.dev.IsolatedDevModeMain] (main) Attempting to start hot replacement endpoint to recover from previous Quarkus startup failure
2021-02-15 14:56:57,676 ERROR [io.qua.run.Application] (Quarkus Main Thread) Failed to start application (with profile dev): java.util.NoSuchElementException: SRCFG00011: Could not expand value ret_status in property PROMPT
	at io.smallrye.config.ExpressionConfigSourceInterceptor.lambda$getValue$0(ExpressionConfigSourceInterceptor.java:44)
	at io.smallrye.common.expression.ExpressionNode.emit(ExpressionNode.java:22)
	at io.smallrye.common.expression.CompositeNode.emit(CompositeNode.java:22)
	at io.smallrye.common.expression.Expression.evaluateException(Expression.java:56)
	at io.smallrye.common.expression.Expression.evaluate(Expression.java:70)
	at io.smallrye.config.ExpressionConfigSourceInterceptor.getValue(ExpressionConfigSourceInterceptor.java:37)
	at io.smallrye.config.ExpressionConfigSourceInterceptor.getValue(ExpressionConfigSourceInterceptor.java:18)
	at io.smallrye.config.SmallRyeConfigSourceInterceptorContext.proceed(SmallRyeConfigSourceInterceptorContext.java:20)
	at io.smallrye.config.SmallRyeConfig.getConfigValue(SmallRyeConfig.java:192)
	at io.quarkus.vertx.http.runtime.devmode.ConfigDescriptionsSupplier.get(ConfigDescriptionsSupplier.java:44)
	at io.quarkus.vertx.http.runtime.devmode.ConfigDescriptionsSupplier.get(ConfigDescriptionsSupplier.java:13)
	at io.quarkus.vertx.http.runtime.devmode.DevConsoleRecorder.addInfo(DevConsoleRecorder.java:37)
	at io.quarkus.deployment.steps.DevConsoleProcessor$runtimeTemplates-308161071.deploy_6(DevConsoleProcessor$runtimeTemplates-308161071.zig:13892)
	at io.quarkus.deployment.steps.DevConsoleProcessor$runtimeTemplates-308161071.deploy(DevConsoleProcessor$runtimeTemplates-308161071.zig:70)
```

**To Reproduce**

Link to a small reproducer (preferably a Maven project if the issue is not Gradle-specific).
Just follow guide of Apache Kafka with reactive messaging like here
https://quarkus.io/guides/kafka

**Configuration**
```properties
## Confluent Cloud Settings
kafka.bootstrap.servers=gcp:9092
kafka.security.protocol=SASL_SSL
kafka.sasl.jaas.config=org.apache.kafka.common.security.plain.PlainLoginModule  \
required username='key'  \
password='password';
kafka.sasl.mechanism=PLAIN
kafka.acks=all

# Required connection configs for Confluent Cloud Schema Registry
kafka.schema.registry.url=https://gcpcloud
kafka.basic.auth.credentials.source=USER_INFO
kafka.basic.auth.user.info=user:password
## Quarkus General

## Quarkus Kafka
# Configure the Kafka sink (we write to it)
mp.messaging.outgoing.generated-price.connector=smallrye-kafka
mp.messaging.outgoing.generated-price.topic=prices
mp.messaging.outgoing.generated-price.value.serializer=org.apache.kafka.common.serialization.IntegerSerializer

# Configure the Kafka source (we read from it)
mp.messaging.incoming.prices.connector=smallrye-kafka
mp.messaging.incoming.prices.value.deserializer=org.apache.kafka.common.serialization.IntegerDeserializer

# Configure the Kafka sink (we write to it)
mp.messaging.outgoing.movies.connector=smallrye-kafka
mp.messaging.outgoing.movies.topic=movies
mp.messaging.outgoing.movies.value.serializer=io.confluent.kafka.serializers.KafkaAvroSerializer
mp.messaging.outgoing.movies.schema.registry.url=https://gcpcloud
mp.messaging.outgoing.movies.specific.avro.reader=true

# Configure the Kafka source (we read from it)
mp.messaging.incoming.movies-from-kafka.connector=smallrye-kafka
mp.messaging.incoming.movies-from-kafka.topic=movies
mp.messaging.incoming.movies-from-kafka.value.deserializer=io.confluent.kafka.serializers.KafkaAvroDeserializer
mp.messaging.incoming.movies-from-kafka.auto.offset.reset=earliest
mp.messaging.incoming.movies-from-kafka.enable.auto.commit=false
mp.messaging.incoming.movies-from-kafka.schema.registry.url=https://gcpcloud
mp.messaging.incoming.movies-from-kafka.specific.avro.reader=true

```

**Screenshots**
(If applicable, add screenshots to help explain your problem.)

**Environment (please complete the following information):**
 - Output of `uname -a` or `ver`: mac OS Bigsur
 - Output of `java -version`: 1.11
 - GraalVM version (if different from Java): 
 - Quarkus version or git rev: 1.11.3
 - Build tool (ie. output of `mvnw --version` or `gradlew --version`): mvnw

**Additional context**
(Add any other context about the problem here.)

**Comment by cescoffier** [comment]
Looks like a config issue. @radcortez any change I should be aware of?

**Comment by radcortez** [comment]
Yes, this is the same issue reported in https://github.com/quarkusio/quarkus/issues/15032.

A fix is already in place: https://github.com/quarkusio/quarkus/pull/15045.

@missourian55 as a workaround, you can set an empty default, like ${something:} in the failing properties.

**Comment by missourian55** [comment]
> Yes, this is the same issue reported in #15032.
> 
> A fix is already in place: #15045.
> 
> @missourian55 as a workaround, you can set an empty default, like ${something:} in the failing properties.

Thanks, the workaround fixes the problem
